src/Module.js

```diff
@@ -24,6 +24,9 @@ class SyntheticDefaultDeclaration {
 	}
 
 	addReference ( reference ) {
+		// Don't change the name to `default`; it's not a valid identifier name.
+		if ( reference.name === 'default' ) return;
```

Comment:
shouldn't this come after `reference.declaration = this`? i.e.

```js
reference.declaration = this;
if ( reference.name !== 'default' ) this.name = reference.name;
```

Or do these types of reference never need to know about their declaration?